app/models/user_permission.rb

```diff
@@ -0,0 +1,13 @@
+class UserPermission < ActiveRecord::Base
+  belongs_to :user
+  validates_presence_of :user_id, :resource_type
+
+  def access_rules
+    rules = super
+    rules && HashWithIndifferentAccess.new(JSON.parse(rules))
```

Comment:
we can eliminate the JSON parse and to_json calls by telling rails to serialize the column as a hash:
``` rb
class UserPermission < ActiveRecord::Base
   serialize :access_rules, Hash
end
```

Reply:
yes lets!